File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.c


int minSubArrayLen(int target, int* nums, int numsSize){
    int l=0,r=0,sum=0,len=INT_MAX;
    while(r<numsSize){
        sum+=nums[r];
        while(sum>=target){
            if(r-l+1<len) len=r-l+1;
            sum-=nums[l++];
        }
        r++;
    }
    if(len==INT_MAX) len=0;
    return len;
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        min_len = float('inf')
        left = 0
        right = 0
        wn_sum = nums[0]
        while left < n:
            while right < n and wn_sum<target:
                right += 1
                wn_sum += nums[right] if right<n else 0

            if right < n:
                min_len = min(min_len, right-left+1)
                
            wn_sum -= nums[left]
            left += 1
        
        min_len = 0 if min_len == float('inf') else min_len

        return min_len